### PR TITLE
Tidy markdown

### PR DESCRIPTION
**Pull request by damacus**
Add yamllint check
Add chefstyle check
Add markdown check
Add spellcheck from chef/chef

Signed-off-by: Dan Webb <dan.webb@damacus.io>

### CODE REVIEW

File: cspell.json [hunk 1]
```diff
@@ -0,0 +1,1862 @@
...
+    "name": "chef",
+    "file": "./chef_dictionary.txt",
+    "description": "Custom Chef Dictionary"
+  }],
+  "dictionaries": ["chef"],
+  "words": [
+    "IOAV",
+    "abcz",
+    "Abdulin",
+    "ABORTIFHUNG",
```
Comment (tas50) [inline cspell.json:14]: Where did this list come from?

File: .github/workflows/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+name: lint
```
[the remark this answers is not in the document]
Comment (damacus) [reply .github/workflows/lint.yaml:2]: I kind of understand this POV, but don't we have lint checks in other places. And from my perspective, hopefully it increases the quality of the markdown in the repo. There are just too many people who don't understand how to write markdown properly, so this at least gives them a helping hand.

**Review by vkarve-chef — APPROVED** [review]

**Review by tas50 — CHANGES_REQUESTED** [review]
I can live with markdown enforcement and yaml enforcement as long as it doesn't impact potential contributions. This does need some cleanup to the cspell config though. Remove all the exceptions and see specifically what we need. Things that are generic computing terms should go in https://github.com/chef/chef_dictionary/ and things that are very specific to this project should go back into the config directly.

**Comment by damacus** [comment]
> I can live with markdown enforcement and yaml enforcement as long as it doesn't impact potential contributions. 

Thankfully we're not requiring yaml or makrdown lint to pass here, so we can merge with errors 👍🏼

**Review by nikhil2611 — APPROVED** [review]

**Pull request merged**